packages/polymarket-notifier/src/polymarketNotifier.js

```diff
@@ -0,0 +1,223 @@
+    this.getTime = getTime;
+    this.apiEndpoint = apiEndpoint;
+    this.maxTimeAfterProposal = maxTimeAfterProposal;
+  }
+
+  // Main function to check recent proposals against Polymarket API data.
+  async checkRecentProposals() {
+    this.logger.debug({
+      at: "PolymarketNotifier",
+      message: "Checking for past proposals",
+      apiEndpoint: this.apiEndpoint,
+      maxTimeAfterProposal: this.maxTimeAfterProposal,
+    });
+
+    const proposalEvents = [];
+    const proposalData = [];
+    const currentTime = await this.getTime();
+    const notifiedProposals = await this.getNotifiedProposals();
+    const questionData = await this.getQuestionData();
+
+    // gets the most updated OO contract
+    const contractAddress = getAddress("OptimisticOracle", 137);
+    const client = uma.clients.optimisticOracle.connect(contractAddress, provider);
+
+    // gets all ProposePrice events using ethers query filter api
+    const events = await client.queryFilter("ProposePrice");
+
+    // creates array for each event
+    events.forEach((request) =>
+      proposalEvents.push({
+        txHash: request.transactionHash,
+        requester: request.args.requester,
+        proposer: request.args.proposer,
+        timestamp: Number(request.args.timestamp),
+        identifier: request.args.identifier,
+        ancillaryData: request.args.ancillaryData,
+        proposedPrice: request.args.proposedPrice.toString(),
+      })
+    );
+
+    // combines data from the Polymarket API data to the proposal event based on ancillaryData
+    for (let i = 0; i < proposalEvents.length; i++) {
+      if (proposalEvents[i].timestamp > currentTime - this.maxTimeAfterProposal) {
+        proposalData.push({
+          ...proposalEvents[i],
+          ...questionData.find((proposalEvent) => proposalEvent.ancillaryData === proposalEvents[i].ancillaryData),
+        });
+      }
+    }
+
+    // checks the proposed price against the Polymarket API data
+    const recentProposals = proposalData
+      .map((contract) => {
+        // excluding proposals without a proposed price and the requester is not Polymarket
+        // the threshold for accepting a proposal is valid is currently set to 0.95 but can be adjusted
+        if (!contract.timestamp || !contract.proposedPrice || contract.requester != polymarketContract) {
+          return null;
+        }
+        // ensures the API price is greater than 0.95 when a 1 is proposed
+        if (contract.proposedPrice === "1000000000000000000" && contract.outcome1Price > 0.95) {
+          return null;
+        }
+        // ensures the API price is greater than 0.95 when a 1 is proposed
+        if (contract.proposedPrice === "0" && contract.outcome2Price > 0.95) {
+          return null;
+        }
+        // the bot currently is not optimized for earlyExpirations but can be updated later
+        if (contract.proposedPrice === earlyExpiryResponse) {
+          return null;
+        }
```

Comment:
```suggestion
        if (contract.proposedPrice === earlyExpiryResponse) return null;
```

might just be a bit cleaner given the logic you have here.